CRM/Event/Form/Registration/Register.php

```diff
@@ -586,6 +591,17 @@ static public function buildAmount(&$form, $required = TRUE, $discountId = NULL)
           if (!is_array($options)) {
             continue;
           }
+          elseif (
+            $hideAdminValues &&
```

Comment:
This could be better as like before the if
```php
$participantEventClasses = array('CRM_Event_Form_Participant', 'CRM_Event_Form_ParticipantFeeSelection');
```
then change 596 and 597 to be 

```php
!in_array($classname, $participantEventClasses)
```